rir/src/compiler/translations/pir_translator.h

```diff
@@ -2,22 +2,24 @@
 #define PIR_TRANSLATOR_H
 
 #include "../pir/module.h"
-#include "../pir/value.h"
-#include "runtime/Function.h"
+#include <string>
 
 namespace rir {
 namespace pir {
 
+class RirCompiler;
 class PirTranslator {
   public:
-    PirTranslator(bool verbose) : verbose(verbose) {}
+    PirTranslator(std::string name)
+        : name(name) {}
 
-    pir::Closure* compileClosure(SEXP);
+    void apply(Closure* function);
+    std::string& getName() { return this->name; }
```

Comment:
no `this`

Reply:
also returning a string reference in a getter is very bad style.

the idea with a reference is that you can modify in place. Ie. a client can change the string with:
`translator->getName() = "another name"` which completely defeats the idea of making `name` private in the first place :)

So either return `const std::string&` which disallows modification of the returned reference. Or return `std::string` (which semantically copies the value) and rely on the C++ compiler optimizations to avoid unnecessary copying.

In this case I personally would opt for the second option.

as a rule of thumb, returning (or passing) non-const references is almost always unwarranted.

Reply:
I Went with the second option. Thanks